src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -1773,6 +1828,93 @@ private static void validateStyleModules(Set<Node> checks, Set<Node> configs,
             Set<String> styleChecks, String styleName, String ruleName) {
         final Iterator<Node> itrChecks = checks.iterator();
         final Iterator<Node> itrConfigs = configs.iterator();
+        final boolean isRegularDocumentation = "google".equals(styleName)
+                && PER_MODULE_TESTS_RULES_LIST.contains(ruleName) || "sun".equals(styleName);
+
+        if (isRegularDocumentation) {
+            while (itrChecks.hasNext()) {
```

Comment:
please move to new method - validateModuleWiseTesting

Reply:
Done.

https://github.com/checkstyle/checkstyle/blob/7e868f5cd0424fa926ed50a7426158666c3941a1/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java#L1835-L1837

https://github.com/checkstyle/checkstyle/blob/7e868f5cd0424fa926ed50a7426158666c3941a1/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java#L1847-L1848